app/jobs/submissions_job.rb

```diff
@@ -43,10 +39,27 @@ def perform(groupings, apply_late_penalty: true, **options)
       grouping.save
       add_warning_messages(grouping.errors.full_messages) if grouping.errors.present?
       progress.increment
+      unless options[:notify_socket].nil? || options[:enqueuing_user].nil?
+        CollectSubmissionsChannel.broadcast_to(options[:enqueuing_user], ActiveJob::Status.get(job_id).to_h)
+      end
     end
+  rescue StandardError => e
+    status.catch_exception(e)
+    raise e
   ensure
     unless options[:notify_socket].nil? || options[:enqueuing_user].nil?
-      CollectSubmissionsChannel.broadcast_to(options[:enqueuing_user], body: 'sent')
+      message = if status&.progress == 1
+                  if status[:warning_message].nil?
+                    { status: :completed }
+                  else
+                    { status: :completed, warning_message:
+                      status[:warning_message] }
+                  end
+                else
+                  ActiveJob::Status.get(job_id).to_h
+                end
+      CollectSubmissionsChannel.broadcast_to(options[:enqueuing_user], message)
+      CollectSubmissionsChannel.broadcast_to(options[:enqueuing_user], update_table: true)
```

Comment:
Seems strange to broadcast two separate messages. You can merge `message` and `{update_table: true}`